src/Symfony/Component/HttpClient/Response/CurlResponse.php

```diff
@@ -277,15 +277,19 @@ private static function select(CurlClientState $multi, float $timeout): int
     /**
      * Parses header lines as curl yields them to us.
      */
-    private static function parseHeaderLine($ch, string $data, array &$info, array &$headers, ?array $options, CurlClientState $multi, int $id, ?string &$location, ?callable $resolveRedirect, ?LoggerInterface $logger): int
+    private static function parseHeaderLine($ch, string $data, array &$info, array &$headers, ?array &$options, CurlClientState $multi, int $id, ?string &$location, ?callable $resolveRedirect, ?LoggerInterface $logger, &$content = null): int
     {
         if (!\in_array($waitFor = @curl_getinfo($ch, CURLINFO_PRIVATE), ['headers', 'destruct'], true)) {
             return \strlen($data); // Ignore HTTP trailers
         }
 
         if ("\r\n" !== $data) {
             // Regular header line: add it to the list
+
             self::addResponseHeaders([substr($data, 0, -2)], $info, $headers);
+            if (isset($headers['content-type'], $options['buffer']) && \is_string($options['buffer'])) {
```

Comment:
this check should be done after, before `curl_setopt($ch, CURLOPT_PRIVATE, 'content');` below, so that we don't check the content-type for intermediary redirections

Reply:
Moved it